### ISSUE TITLE
Disable and enable automatic event logging arcording to GDPR

### ISSUE THREAD

**Issue by Hanspagh** [facebook/react-native-fbsdk#465]
Hi. I can see it is possible in the android and iOS sdk to disable and then enable automatic event logging again after user consent see the provided [Link](https://developers.facebook.com/ads/blog/post/2018/05/10/compliance-protections-gdpr/). Is it possible to do the same in the RN sdk?

**Comment by peteychuk** [comment]
Any updates on this?

**Comment by Hanspagh** [comment]
I ended up fixing it by editing the info.Plist and the android manifest of the facebook SDK as described here https://developers.facebook.com/docs/app-events/getting-started-app-events-android#disable-auto-events and then made a custom hook to enable them

**Comment by peteychuk** [comment]
Thank you @Hanspagh,
For IOS documentation described here:
https://developers.facebook.com/docs/app-events/getting-started-app-events-ios (Disable Automatically Logged Events)

I think we can resolve this issue.

**Comment by obendevski** [comment]
Any updates regarding this?

**Comment by obendevski** [comment]
> I ended up fixing it by editing the info.Plist and the android manifest of the facebook SDK as described here https://developers.facebook.com/docs/app-events/getting-started-app-events-android#disable-auto-events and then made a custom hook to enable them

Can you please tell me how you did it with the custom hook?

**Comment by Hanspagh** [comment]
@obendevski Its been a while, but it was something a long the lines of this
https://dev.to/evenmik/react-native-bridge-for-android-and-ios-405k
and then the native call was just to `setAutoLogAppEventsEnabled(true)`

**Comment by obendevski** [comment]
> @obendevski Its been a while, but it was something a long the lines of this
> https://dev.to/evenmik/react-native-bridge-for-android-and-ios-405k
> and then the native call was just to `setAutoLogAppEventsEnabled(true)`

Great, thank you very much. Just one more thing, where do you import  the `setAutoLogAppEventsEnabled ` from ? I will need to install the separate sdks for android and iOS, right? Because I cant find any reference to the method in the `react-native-fb-sdk`.

**Comment by Hanspagh** [comment]
Yes, but I am also pretty sure, this is a requiment for the libary anyways

**Comment by stephanpavlovic** [comment]
> > @obendevski Its been a while, but it was something a long the lines of this
> > https://dev.to/evenmik/react-native-bridge-for-android-and-ios-405k
> > and then the native call was just to `setAutoLogAppEventsEnabled(true)`
> 
> Great, thank you very much. Just one more thing, where do you import the `setAutoLogAppEventsEnabled ` from ? I will need to install the separate sdks for android and iOS, right? Because I cant find any reference to the method in the `react-native-fb-sdk`.

Did you pull it of @obendevski? I'm trying to do the exact same thing ...

**Comment by obendevski** [comment]
@stephanpavlovic I did it, yes. I will write it here ASAP, next few days.

**Comment by plaa** [comment]
@obendevski I am struggling with importing classes from the react-native-fbsdk libraries.  How did you handle them?

I managed to get iOS working by adding `#import <FBSDKCoreKit.h>` to my Swift bridging header and then using `Settings.isAutoLogAppEventsEnabled = enabled`.

On Android I am unable to `import com.facebook.FacebookSdk`. It seems that the library is included through react-native-fbsdk, but is not on the classpath during build time.  I even tried [manually linking the package](https://github.com/facebook/react-native-fbsdk#2-link) (only gradle files) but compiling still fails.

**Comment by obendevski** [comment]
Hey, @plaa , I had the same problem, had to add the android fbsdk.
In app/build.gradle add: 
`implementation 'com.facebook.android:facebook-android-sdk:[5,6)'`.
In android/build.gradle, add:
`mavenCentral()` in `repositories`.

Then, you can `import com.facebook.FacebookSdk;` in your custom module.
Hope it helps,
Let me know if it works!

**Comment by plaa** [comment]
Thanks @obendevski!

For me it was sufficient to add the `implementation` statement in app/build.gradle, the `mavenCentral()` was unnecessary. I guess the npm packages add that automatically, as it's not in my config.  I had to exit + clean to get Android Studio to detect the library.

For those wondering, the `implementation` statement requires any 5.x version of facebook-android-sdk to be included in the build. The react-native-fbsdk npm package will decide the exact version that will be included.
